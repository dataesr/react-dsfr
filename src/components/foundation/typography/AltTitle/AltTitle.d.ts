import * as React from 'react';

export type AltTitleChildren = string | React.ReactNode;

export type AltTitleClassName = string | Object | any[];

export type AltTitleAs = "h1" | "h2" | "h3" | "h4";

export type AltTitleSize = "xl" | "lg" | "md" | "sm" | "xs";

export interface AltTitleProps {
    children: AltTitleChildren;
    className?: AltTitleClassName;
    /**
     * html tag to render
     */
    spacing?: string;
    as: AltTitleAs;
    size: AltTitleSize;
}

declare const AltTitle: React.FC<AltTitleProps>;

export default AltTitle;

