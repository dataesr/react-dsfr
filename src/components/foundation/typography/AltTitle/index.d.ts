import type AltTitle from './AltTitle';

export type { AltTitle };
