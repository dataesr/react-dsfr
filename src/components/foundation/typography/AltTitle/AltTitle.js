import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { getSpace } from '../../../../utils/getters';

/**
 *
 * @visibleName AltTitle
 */
const AltTitle = ({
  as, size, children, spacing, className,
}) => {
  const HtmlTag = `${as}`;
  const { margin, padding } = getSpace(spacing);

  const _className = classNames(className, padding, margin, { [`fr-display--${size}`]: size });
  return (
    <HtmlTag className={_className}>
      {children}
    </HtmlTag>
  );
};

AltTitle.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  /**
     * html tag to render
     */
  spacing: PropTypes.string,
  as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']).isRequired,
  size: PropTypes.oneOf(['xl', 'lg', 'md', 'sm', 'xs']).isRequired,
};

AltTitle.defaultProps = {
  className: '',
  spacing: '',
};

export default AltTitle;
