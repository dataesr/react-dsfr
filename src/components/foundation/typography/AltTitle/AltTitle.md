### Alternative titles

```jsx
<AltTitle as="h1" size="xl">Title xl</AltTitle>
```

```jsx
<AltTitle as="h1" size="lg">Title lg</AltTitle>
```

```jsx
<AltTitle as="h1" size="md">Title md</AltTitle>
```

```jsx
<AltTitle as="h1" size="sm">Title sm</AltTitle>
```

```jsx
<AltTitle as="h1" size="xs">Title xs</AltTitle>
```
