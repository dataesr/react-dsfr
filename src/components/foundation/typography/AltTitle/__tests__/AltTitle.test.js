import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import AltTitle from '../AltTitle';

const initialProps = {
  as: 'h2',
  size: 'xl',
  children: 'Title Alt xl',
};

describe('<AltTitle />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(<AltTitle {...initialProps}>{initialProps.children}</AltTitle>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
