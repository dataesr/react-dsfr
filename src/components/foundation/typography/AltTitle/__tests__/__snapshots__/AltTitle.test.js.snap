// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AltTitle /> renders correctly 1`] = `
<h2
  className="fr-display--xl"
>
  Title Alt xl
</h2>
`;
